web/packages/shared/components/UnifiedResources/ListView/ResourceListItem.tsx

```diff
@@ -0,0 +1,321 @@
+import styled from 'styled-components';
+
+import { Box, ButtonIcon, Flex, Label, Text } from 'design';
+import { StyledCheckbox } from 'design/Checkbox';
+import { Tags } from 'design/Icon';
+import { ResourceIcon } from 'design/ResourceIcon';
+
+import { makeLabelTag } from 'teleport/components/formatters';
+
+import { HoverTooltip } from 'shared/components/ToolTip';
+
+import { ResourceItemProps } from '../types';
+import { PinButton } from '../shared/PinButton';
+import { CopyButton } from '../shared/CopyButton';
+
+export function ResourceListItem({
+  name,
+  primaryIconName,
+  SecondaryIcon,
+  onLabelClick,
+  listViewProps,
+  ActionButton,
+  labels,
+  pinningSupport,
+  pinned,
+  pinResource,
+  selectResource,
+  selected,
+}: ResourceItemProps) {
+  const { description, type, addr } = listViewProps;
+
+  const [showLabels, setShowLabels] = useState(false);
+  const [hovered, setHovered] = useState(false);
+
+  return (
+    <RowContainer
+      onMouseEnter={() => setHovered(true)}
+      onMouseLeave={() => setHovered(false)}
+    >
+      <RowInnerContainer alignItems="start" pinned={pinned} selected={selected}>
+        {/* checkbox */}
+        <HoverTooltip
+          css={`
+            grid-area: checkbox;
+          `}
+          tipContent={selected ? 'Deselect' : 'Select'}
+        >
+          <StyledCheckbox checked={selected} onChange={selectResource} />
+        </HoverTooltip>
+
+        {/* pin button */}
+        {/* We wrap it in a box in order to center it properly in its grid. */}
+        <Box
+          css={`
+            grid-area: pin;
+            place-self: center center;
+          `}
+        >
+          <PinButton
+            setPinned={pinResource}
+            pinned={pinned}
+            pinningSupport={pinningSupport}
+            hovered={hovered}
+          />
+        </Box>
+
+        {/* icon */}
+        <ResourceIcon
+          name={primaryIconName}
+          width="36px"
+          height="36px"
+          css={`
+            grid-area: icon;
+            place-self: center center;
+          `}
+        />
+
+        {/* name and description */}
+        <Flex
+          css={`
+            grid-area: name;
+            justify-content: left;
+            align-items: center;
+            overflow: hidden;
+          `}
+        >
+          <Flex
+            flexDirection="column"
+            css={`
+              overflow: hidden;
+            `}
+          >
+            <HoverTooltip tipContent={name} showOnlyOnOverflow>
+              <Name>{name}</Name>
+            </HoverTooltip>
+            <HoverTooltip tipContent={description} showOnlyOnOverflow>
+              <Description>{description}</Description>
+            </HoverTooltip>
+          </Flex>
+          <Box
+            css={`
+              align-self: start;
+            `}
+          >
+            {hovered && <CopyButton name={name} ml={1} />}
+          </Box>
+        </Flex>
+
+        {/* type */}
```

Comment:
You know what would be cool? If we could show the full type (without ellipsis) if there is no address.
<img width="1046" alt="image" src="https://github.com/gravitational/teleport/assets/20583051/2c1e0900-3524-4133-a1cd-44ffef5b89a8">
Now we are wasting space a bit :)